### DISCUSSION TITLE
Site problem with the new google-ima shim script

### DISCUSSION THREAD

**Discussion by vlaszlo** [uBlockOrigin/uBlock-issues#2264] [Q&A] [answered]
STR (using the most current Firefox nightly, if it matters):
1. Create a new profile, install 1.44.3b3.
2. Important: Purge and update filter lists.
3. On [EHF TV](https://ehftv.com/player/vod/metz-handball-v-storhamar-handball-elite-match-highlights-r1?entryId=0_6lco8i4c&competitions=EHF%20Champions%20League%20Women&categories=Highlights), click through 2 or 3 videos, you don't actually have to watch them to the end. Notice they are functioning normally.
4. Install 1.44.3b4.
5. Go to [EHF TV](https://ehftv.com/player/vod/metz-handball-v-storhamar-handball-elite-match-highlights-r1?entryId=0_6lco8i4c&competitions=EHF%20Champions%20League%20Women&categories=Highlights), try to watch a few videos again. They now either don't load at all, or they play with sound but no picture.

If I interpret the logger right, the google-ima script is blocked with beta3, while it's replaced with the shim in beta4. Allowing the connection to imasdk.googleapis.com makes the videos work normally again in beta4, too.

Now the question: Filter issue (an exception needs to be added) or uBlock issue (the shim script needs to be improved)?

**Comment by mapx-** [comment]
@gorhill

**Comment by gorhill** [comment, marked as answer]
I can reproduce, it looks like some timing issue, I get the player working fine if I toggle off the autoplay button top right of the page, and eventually it works fine after reloading the page regardless of the state of that button -- though I have to click play for the video to start. Need more investigation.

**Reply by gorhill** [reply]
Though this needs more investigation, for now can you try:

    ehftv.com##.blackPlayer

The video is playing, it's just that its covered with a black rectangle.

**Reply by vlaszlo** [reply]
> ehftv.com##.blackPlayer

That works, yes. Thanks.

**Reply by gorhill** [reply]
I don't have time now to get to the bottom of this so I added the cosmetic filter to Unbreak list.

**Reply by vlaszlo** [reply]
OK. Thanks again.
